### PR TITLE
Check participants listed in phenotype/*.tsv and subjects list 

### PR DESCRIPTION
**Pull request by suyashdb**
Found that current script did not check correctly(it was a bug) participants listed in `phenotype/*.tsv` and subjects in dataset.

### CODE REVIEW

**Comment by suyashdb** [comment]
I will write tests for this check as well. 
As test was not written previously even though this check was not working correctly until now, it never got corrected.

**Comment by suyashdb** [comment]
ready for review.

File: validators/tsv.js [hunk 1]
```diff
@@ -142,6 +145,27 @@ module.exports = function TSV (file, contents, fileList, callback) {
...
+    //                     phenotypeParticipants.push({
+    //                         list: participantList,
+    //                         file: file
+    //                     });
+    //                 }
+    //             }
+    //             // self.issues = self.issues.concat(issues);
+    //             process.nextTick(cb);
+    //         });
+    //     });
```
Comment (chrisgorgo) [inline validators/tsv.js]: Please remove.

File: validators/bids.js [hunk 2]
```diff
@@ -422,19 +422,10 @@ BIDS = {
...
-                            self.issues.push(new Issue({
-                                code: 51,
-                                evidence: 'sub-' + diff.join(', sub-'),
-                                file: fileParticpants.file
-                            }));
-                        }
-                    }
-                }
+                //check for equal number of participants from ./phenotype/*.tsv and participants in dataset
+                TSV.checkphenotype_tsv(phenotypeParticipants, summary, self.issues);
```
Comment (chrisgorgo) [inline validators/bids.js]: You can do similar thing for participant.tsv

**Comment by suyashdb** [comment]
In codeClimate "Function TSV has a Cognitive Complexity of 55 (exceeds 5 allowed). Consider refactoring" need to be fixed in a different PR. an issue has been created. 

Ready for review.

File: tests/tsv.spec.js [hunk 3]
```diff
@@ -89,24 +89,50 @@ describe('TSV', function(){
...
+                [
+                    '10159',
+                    '10171'
+                ],
+            tasks: [ ],
+            totalFiles: 43,
+            size: 11845 };
+        var issues = [];
+        validate.TSV.checkphenotype(phenotypeParticipants, summary, issues, function(issues){
+            assert(issues.length === 0 && issues[0].code === 51);
```
Comment (chrisgorgo) [inline tests/tsv.spec.js]: How is this ever true? accessing first element from zero element list?

**Comment by chrisgorgo** [comment]
Is this ready for review?

**Comment by suyashdb** [comment]
@chrisfilo: yes this is ready for review.

**Pull request merged**